testing/regress/ecl/spray_dir_test.ecl

```diff
@@ -0,0 +1,217 @@
+    self.failifnosourcefile := l.failifnosourcefile;
+    self.result := l.result;
+end;
+
+
+dst2 := NOFOLD(DATASET([{emptyDirPath+'/*', false, 'Pass', ''}], sprayrec));
+p2 := PROJECT(NOFOLD(dst2), spray(LEFT));
+c2 := CATCH(NOFOLD(p2), ONFAIL(TRANSFORM(sprayrec,
+                                SELF.sourcepath := emptyDirPath+'/*',
+                                SELF.failifnosourcefile := false,
+                                SELF.result := 'Fail',
+                                SELF.msg := FAILMESSAGE
+                                )));
+#if (VERBOSE = 1)
+    sprayOut1 := output(c2);
+#else
+    sprayOut1 := output(c2, {result}, NAMED('Empty_Dir_No_FailIFNoSourceFile'));
+#end
+
+
+// Try to spray from an empty directory with failifnosourcefile=1 
+// It should always fail so if it pass something broken
+
+dst3 := NOFOLD(DATASET([{emptyDirPath+'/*', true, 'Fail', ''}], sprayrec));
+p3 := PROJECT(NOFOLD(dst3), spray(LEFT));
+c3 := CATCH(NOFOLD(p3), ONFAIL(TRANSFORM(sprayrec,
+                                 SELF.sourcepath := emptyDirPath+'/*',
+                                 SELF.failifnosourcefile := true,
+                                 SELF.result := 'Pass',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+#if (VERBOSE = 1)
+    sprayOut2 := output(c3);
+#else
+    sprayOut2 := output(c3, {result}, NAMED('Empty_Dir_FailIFNoSourceFile'));
+#end
+
+
+// Try to spray from a not empty directory with default failifnosourcefile (=0) 
+
+dst4 := NOFOLD(DATASET([{notEmptyDirPath+'/*', false, 'Pass', ''}], sprayrec));
+p4 := PROJECT(NOFOLD(dst4), spray(LEFT));
+c4 := CATCH(NOFOLD(p4), ONFAIL(TRANSFORM(sprayrec,
+                                 SELF.sourcepath := notEmptyDirPath+'/*',
+                                 SELF.failifnosourcefile := false,
+                                 SELF.result := 'Fail',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+#if (VERBOSE = 1)
+    sprayOut3 := output(c4);
+#else
+    sprayOut3 := output(c4, {result}, NAMED('Not_Empty_Dir_No_FailIFNoSourceFile'));
+#end
+
+
+// Try to spray from a not empty directory with failifnosourcefile=1 
+
+dst5 := NOFOLD(DATASET([{notEmptyDirPath+'/*', true, 'Pass', ''}], sprayrec));
+p5 := PROJECT(NOFOLD(dst5), spray(LEFT));
+c5 := CATCH(NOFOLD(p5), ONFAIL(TRANSFORM(sprayrec,
+                                 SELF.sourcepath := notEmptyDirPath+'/*',
+                                 SELF.failifnosourcefile := true,                                 
+                                 SELF.result := 'Fail',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+#if (VERBOSE = 1)
+    sprayOut4 := output(c5);
+#else
+    sprayOut4 := output(c5, {result}, NAMED('Not_Empty_Dir_FailIFNoSourceFile'));
+#end
+
+
+SEQUENTIAL(
+    setupFile,
+    desprayOut,
+    sprayOut1,
+    sprayOut2,
+    sprayOut3,
+    sprayOut4,
```

Comment:
probably should also add something here to check the ones that are supposed to have succeeded in creating a logical file (sprayOut1 and sprayOut3), actually have, e.g. perhaps a COUNT(<logicalfile>) , which would fail if spray was missing and key file can check correct result.

Reply:
Right, good point.